Tutorials pages

smallbrownbike opened the pull request:
## Changes

- Moves tutorials from `/docs/tutorials` to `/tutorials`
- Programmatically creates pages for tutorial contributors and categories (`/tutorials/contributors/james-hawkins`, `/tutorials/categories/funnels`)
- Filters for authors and categories

  - Gatsby is so fast you don't even notice these are actually navigating to different pages!
- Pulls in page views at build time _from_ PostHog and lists them on individual tutorial pages

  - This is cool but not quite perfect. Pulling from a shared dashboard doesn't allow us to get all of the pages just yet. When no page views are present, the section isn't shown.
- Supports multiple design view types (card, list) that persist when refreshing the page
- Dark mode is almost implied now 😎, but still worth noting

TLDR: New tutorials page(s) based on [this design](https://www.figma.com/proto/g1UCTW9ZX21GU85hLsWB31/Website-concepts?page-id=2055%3A37159&node-id=2198%3A42119&viewport=308%2C48%2C1&scaling=min-zoom&starting-point-node-id=2198%3A42119&show-proto-sidebar=1).

jamesefhawkins commented:
This is so great.

Feedback: I felt myself wanting to "unselect" to view all of these easily. Could we make it so if I click feature flags like this:

<img width="1070" alt="Screenshot 2021-12-09 at 13 09 12" src="https://user-images.githubusercontent.com/47497682/145402352-fbbafa77-dbac-4d5a-b51a-ec194f6dfa43.png">

That i could click 'feature flags' again to unselect/view the whole list?

Also feels like the tutorials link at top left should take you back to posthog.com/tutorials

<img width="482" alt="Screenshot 2021-12-09 at 13 09 51" src="https://user-images.githubusercontent.com/47497682/145402445-3fe4760a-b844-47d1-bc43-5fd53be81529.png">

smallbrownbike commented:
@jamesefhawkins Great feedback thank you! Fixed.

corywatilo commented:
Looking so good!

I've made a few tweaks in the mocks to clean up a few of the weaker aspects to these designs.

### 1. [Index view filtering](https://www.figma.com/file/g1UCTW9ZX21GU85hLsWB31/Website-concepts?node-id=2922%3A78067)

- I've reused the subnav that we're already using on the [product detail pages](https://posthog.com/product/funnels).
- We should limit to only one section being opened at a time. (If we allow multiple to be open, you might think you can select a filter from each section, where you can only filter by one, thus eliminating this confusion.)
- The style of the expandable/collapsible headers in this subnav are what I'd like to run with in other sections in the future, so this will be reused.
    1. Current/active section has background.
    2. Hovering inactive section headers would show the background. 
    3. Normal state: no background on subheaders
    4. A section header will never be the currently-active item - only an item contained within

### 2. [Index: list view](https://www.figma.com/file/g1UCTW9ZX21GU85hLsWB31/Website-concepts?node-id=2922%3A80602)

This design didn't feel quite polished, so I updated in this mock.

### 3. Tutorial article view changes

**Tags**

Currently, tags they look like this:

![image](https://user-images.githubusercontent.com/154479/145532487-ea96eb96-912b-4de8-b3ce-589e48ad5063.png)

**Changes to tags**

1. Let's make tags inline, so multiple tags can appear on a single row if there's space
2. They're just a little big, so we should create a small version of these chips.
    1. Reduce tag font size by ~2px
    2. Possibly reduce padding too (will need to play with this a bit)

**Other**

- [ ] Let's make sure og:image is using the poster image? (We should ultimately remove all poster images from the post themselves, and use the image in the open graph preview
- [ ] Other content pages have a `.article-content` div which applies font sizing, line height, and an animated anchor link effect. We should keep it consistent here.
- [ ] Should Facebook share open in a new window? Looks a little weird for it to open in a full, new window since that modal was designed to open in a popup
- [ ] ☝️ Same with LinkedIn?
- [ ] Email share: Not that anyone will ever use this, but how about we make the post title the subject, and send the URL to the body?

### Open questions

1. Should we remove `/categories` from the slug? So...

    Current: 
    `/tutorials/categories/funnels`

    Proposed:
    `/tutorials/funnels`

2. We also still need to plan how to handle moving Tutorials _out_ of Docs, how we link them from Docs, add them to the Docs dropdown (for now), and add a footer link to this new section

smallbrownbike commented:
@corywatilo Thank you! All of those changes have been made.

## Answer to Question 1
I did it this way for a couple of reasons.
- `/tutorials/${category}` would cause URL conflicts in some cases. For instance, we already have a tutorial at `/tutorials/feature-flags`.

- Adding the filter type to the URL gives more context and feels more SEO-friendly. `/tutorials/categories/${category}` and `/tutorials/contributors/${contributor}` is more descriptive than `/tutorials/${category}` and `/tutorials/${contributor}`

- I think it would be cool to eventually have a page that lists all of the categories and contributors in one place. Giving the URLs this structure will set us up for that.

## Answer to Question 2
I added a redirect for all old `/docs/tutorials` links for now. I also added Tutorials to the footer under Getting started. This feels like the right place but happy to move it.

## Answer to Question 3
I submitted a PR to get `/tutorials` indexed! 🎉

## Answer to Question 4
**ABSOLUTELY NOT** <sub><sup>_but secretly yes it's my favorite shhh_</sup></sub>

corywatilo commented:
- [ ] On mobile, let's move this to the bottom and give it a section header: _"Filter by..."_ and make sure it's wider to align with content
    ![image](https://user-images.githubusercontent.com/154479/145851310-31a4f373-1c72-42a3-83f0-04cf58f985b2.png)

- [ ] Singular _"Contributor_" unless `>1` contributor?
    ![image](https://user-images.githubusercontent.com/154479/145851591-73f8e06b-e6b9-478a-b3aa-aedac73824ee.png)

- [ ] Chips look a little large all around. Can we reduce padding/size?
    ![image](https://user-images.githubusercontent.com/154479/145851894-ad89324a-7ec0-4895-ad30-67364ecff2a2.png)

- [ ] Change link colors to match other mock
    Current:
    ![image](https://user-images.githubusercontent.com/154479/145852013-27f46e94-6c8a-4ecf-886c-4b39eb79f676.png)
    Expected:
    ![image](https://user-images.githubusercontent.com/154479/145852071-da39e33b-dd71-46ea-97c0-121ae78adf9e.png)

corywatilo reviewed (approved):
So excited about this!
